idaes/core/util/model_serializer.py

```diff
@@ -262,34 +252,41 @@ def __init__(
             "ub": _set_ub,
             "value": _set_value,
         }
-        # Add skip classes to classes list, with None as attr list to skip
-        skip_classes2 = []  # need to put skips at front of list
-        self.classes = [i[0] for i in classes]
-        for i in skip_classes:
-            if i not in self.classes:
-                skip_classes2.append((i, None))
-        classes = skip_classes2 + list(classes)  # comined classes with skips
-        # Create lists of classes, attribute lists, and filter functions
-        # Can get class index from class list the use it to get associated items
-        self.classes = [i[0] for i in classes]
+        skip_classes = []
+        # Convert to lists, so we can add things
+        classes = list(classes)
+        data_classes = list(data_classes)
+        # Block and BlockData are required for model structure
+        if Block not in classes:
+            classes.append((Block, ()))
+        if Block._ComponentDataClass not in data_classes:
+            data_classes.append((Block._ComponentDataClass, ()))
```

Comment:
Out of curiosity, why are these lists and not dicts?  The `in` here is a linear scan of the list and could be quite slow.  A dict would support faster lookup and avoid ambiguity if the class appears more than once in the list.

Reply:
I wrote this a very long time ago, before normal dictionaries preserved order.  Since the order of the classes is important, I guess I just went with lists or tuples.  It's been so long I don't have a clear memory of why I did what I did.  I may take another look at this when I fix the typo.  Dicts are at least more aesthetically pleasing.  I did think about this being slow, but this is in a class that's pretty simple and gets created up front, so the efficiency of this methods won't have any measurable impact on overall speed.